src/main/java/org/mule/extension/http/internal/listener/HttpListener.java

```diff
@@ -233,6 +250,31 @@ private void sendErrorResponse(HttpListenerErrorResponseBuilder errorResponse,
     responseCallback.responseReady(response, getResponseFailureCallback(responseCallback, completionCallback));
   }
 
+  private void sendBackPressureResponse(BackPressureContext ctx, SourceCompletionCallback completionCallback) {
```

Comment:
In theory, there was already logic somewhere sending  `SERVICE_UNAVAILABLE` but it doesn't seem you have reused it or deleted it either.

Reply:
The logic seems to be in `createFailureResponseBuilder`

Reply:
this logic is similar to createFailureResponseBuilder() but with significant differences. Notice that I don't have a base error builder, the Error object will not have a Message associated with it, etc.

Reply:
We need to figure what of that won't happen now and remove that code.

Reply:
I think it's just remove:
```
if (error.getErrorType().equals(sourceOverloadErrorType)) {
         failureResponseBuilder = createDefaultFailureResponseBuilder(error, SERVICE_UNAVAILABLE);
       } 
```